### PR TITLE
 Implemented functionality to ignore the path when updating the search path

### PR DESCRIPTION
**Pull request by vsantos-desenv**
Implemented functionality to ignore the path when updating the search path.

Just set the ignorepath property in boss.json as an example below.

```
"ignorepath": [
	"test",
	"tests",
	"samples"
]
```

### CODE REVIEW

File: utils/librarypath/librarypath.go [hunk 1]
```diff
@@ -39,6 +40,25 @@ func cleanPath(paths []string, fullPath bool) []string {
 	return processedPaths
 }
 
+func ExistsInArray(array []string, item string) bool {
```
Comment (snakeice) [inline utils/librarypath/librarypath.go:43]: You can use https://github.com/HashLoad/boss/blob/ed9aae1b2d3a9583e62629ae58e023153629c98c/utils/arrays.go#L5

File: utils/librarypath/librarypath.go [hunk 2]
```diff
@@ -39,6 +40,25 @@ func cleanPath(paths []string, fullPath bool) []string {
...
+	for _, value := range array {
+		if strings.Contains(strings.ToLower(item), strings.ToLower(value)) {
```
Comment (snakeice) [inline utils/librarypath/librarypath.go:45]: Using `contains` in this case invalidate paths:
```go
{
  "ignorePath": [
	"vendor"
  ]
}
```

To path `./vendors` is valid but the path `./src/vendor/vendorCheck.pas` is ignored

File: utils/librarypath/dprojUtil.go [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
+	"github.com/hashload/boss/models"
+	"github.com/hashload/boss/msg"
+	"github.com/hashload/boss/utils"
 )
 
 func updateDprojLibraryPath(pkg *models.Package) {
 	var dprojsNames = GetDprojNames(pkg)
 	for _, dprojName := range dprojsNames {
-		updateLibraryPathProject(dprojName)
+
```
Comment (snakeice) [inline utils/librarypath/dprojUtil.go:22]: Remove this empty line

File: utils/librarypath/librarypath.go (hunk 2 above)
Comment (vsantos-desenv) [reply utils/librarypath/librarypath.go:45]: Do you have any solution for this problem? And if we put back slash before and after the path name in the boos.json file. Can you see any problems?

```
{
  "ignorePath": [
	"\vendor\"
  ]
}
```

Comment (snakeice) [reply utils/librarypath/librarypath.go]: I think using strings.HasPrefix is better for now, based on the relative path of the project folder

File: utils/librarypath/librarypath.go [hunk 4]
```diff
@@ -39,6 +40,21 @@ func cleanPath(paths []string, fullPath bool) []string {
 	return processedPaths
 }
 
+func RemoveIgnoredPath(paths []string, ignoredPaths []string) []string {
+	var result []string
+	result = utils.Filter(paths, func(s string) bool {
+		var result bool = false
+		for _, value := range ignoredPaths {
+			result = strings.Contains(strings.ToUpper(s), strings.ToUpper(value))
```
Comment (snakeice) [inline utils/librarypath/librarypath.go:48]: I think using strings.HasPrefix is better for now, based on the relative path of the project folder.

**Pull request closed without merging**